commit 4510414f158b709529d3e6463dd31bcfab651795
Author: Max Horn <max@quendi.de>
Date:   2022-03-01 00:24:13 +0100

    Use GAP's gap_all.h
    
    Note that it is C++ safe, so remove old workarounds

diff --git a/gap_cpp_headers/gap_cpp_mapping.hpp b/gap_cpp_headers/gap_cpp_mapping.hpp
index d1bfb50..099ccef 100644
--- a/gap_cpp_headers/gap_cpp_mapping.hpp
+++ b/gap_cpp_headers/gap_cpp_mapping.hpp
@@ -8,9 +8,8 @@
 #include <deque>
 #include <list>
 
-// We have to include this to get around problems with the 'extern C' wrapping of compiled.h,
-// which includes gmp, which in C++ mode has some C++ templates.
-#include "include_gap_headers.hpp"
+#include "gap_all.h"
+
 #include "gap_exception.hpp"
 #include "gap_function.hpp"
 
diff --git a/gap_cpp_headers/gap_function.hpp b/gap_cpp_headers/gap_function.hpp
index 99051ac..c6ab6fd 100644
--- a/gap_cpp_headers/gap_function.hpp
+++ b/gap_cpp_headers/gap_function.hpp
@@ -2,7 +2,7 @@
 #define GAP_FUNCTION_HPPQR
 
 #include <string>
-#include "include_gap_headers.hpp"
+#include "gap_all.h"
 
 class GAPFunction
 {
diff --git a/gap_cpp_headers/include_gap_headers.hpp b/gap_cpp_headers/include_gap_headers.hpp
deleted file mode 100644
index 9fca573..0000000
--- a/gap_cpp_headers/include_gap_headers.hpp
+++ /dev/null
@@ -1,11 +0,0 @@
-#ifndef _GAP_HEADERS_H_QEHJ
-#define _GAP_HEADERS_H_QEHJ
-
-#include <stdlib.h>
-#include <stddef.h>
-#include "gmp.h"
-extern "C" {
-#include "compiled.h"
-}
-
-#endif
diff --git a/src/ferret.cc b/src/ferret.cc
index 7b7fa74..f3d6e7d 100644
--- a/src/ferret.cc
+++ b/src/ferret.cc
@@ -3,7 +3,7 @@
 ****************************************************************************/
 
 #include <stdio.h>
-#include "include_gap_headers.hpp"
+#include "gap_all.h"
 #include "c_to_cpp.h"
 #include "generated_headers/function_objs.h"
 
